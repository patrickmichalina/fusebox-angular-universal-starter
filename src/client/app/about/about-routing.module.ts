import { AboutComponent } from './about.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { MetaGuard } from '@ngx-meta/core'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: AboutComponent,
        canActivate: [MetaGuard],
        data: {
          meta: {
            title: 'i18n.about.title',
            description: 'i18n.about.description'
          },
          response: {
            cache: {
              directive: 'no-cache'
            }
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class AboutRoutingModule { }
