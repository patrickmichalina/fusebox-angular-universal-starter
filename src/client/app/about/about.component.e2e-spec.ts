import { browser } from 'protractor';

describe('About', () => {

  beforeEach(async () => {
    return await browser.get('/about');
  });

  it('should have correct feature heading', () => {
    expect(browser.getTitle()).toEqual('About - Fusebox Angular Universal Starter');
  });

});