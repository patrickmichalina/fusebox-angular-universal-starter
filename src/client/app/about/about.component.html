<h1>About</h1>
<p>An Angular starter with all the tools necessary to create server rendered Angular applications.</p>

<form [formGroup]="form" novalidate>
  <mat-form-field>
    <input matInput placeholder="Title" formControlName="title">
  </mat-form-field>
  <mat-form-field>
    <textarea matInput placeholder="html" formControlName="html"></textarea>
  </mat-form-field>
  <button mat-raised-button color="primary" (click)="create()" [disabled]="!form.valid">
    Post
  </button>
  <button mat-raised-button color="warn" (click)="removeAll()">
    Delete All Posts
  </button>
</form>

<mat-card *ngFor="let post of posts$ | async; trackBy: trackByPost">
  <h4>{{ post.title }}</h4>
  <div [innerHTML]="post.html"></div>
</mat-card>