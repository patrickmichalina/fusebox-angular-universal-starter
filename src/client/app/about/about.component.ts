import { PlatformService } from './../shared/services/platform.service'
import { ChangeDetectionStrategy, Component } from '@angular/core'
import { FirebaseDatabaseService } from '../shared/services/firebase-database.service'
import { FormControl, FormGroup, Validators } from '@angular/forms'

@Component({
  selector: 'pm-about',
  templateUrl: './about.component.html',
  styleUrls: ['./about.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AboutComponent {
  readonly posts$ = this.db.getList<{ readonly title: string, readonly html: string }>('posts').map(a => {
    return a.map((b: any) => {
      return {
        title: b.title || '',
        html: b.html || ''
      }
    })
  })

  public readonly form = new FormGroup({
    title: new FormControl('', [
      Validators.required
    ]),
    html: new FormControl('', [
      Validators.required
    ])
  })

  constructor(private db: FirebaseDatabaseService, ps: PlatformService) { }

  create() {
    this.db.getListRef('posts').push(this.form.value)
  }

  removeAll() {
    this.db.getObjectRef('posts').remove()
  }

  trackByPost(index: number, item: any) {
    return index
  }
}
