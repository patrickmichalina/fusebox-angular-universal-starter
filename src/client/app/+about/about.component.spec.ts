import { AboutComponent } from './about.component';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Component } from '@angular/core';
import { AboutModule } from './about.module';

describe(AboutComponent.name, () => {
  let fixture: ComponentFixture<AboutComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [AboutModule],
      declarations: [TestComponent]
    }).compileComponents();
  }));

  beforeEach(async(() => {
    fixture = TestBed.createComponent(AboutComponent);
  }));

  afterEach(async(() => {
    TestBed.resetTestingModule();
  }));

  it('should match snapshot', async(() => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  }));

  it('should compile', async(() => {
    fixture.detectChanges();
    expect(fixture.nativeElement).toBeDefined();
  }));
});

@Component({
  selector: 'test-component',
  template: '<pm-about></pm-about>'
})
class TestComponent {}
