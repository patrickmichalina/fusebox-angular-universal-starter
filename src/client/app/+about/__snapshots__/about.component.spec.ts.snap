// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent should match snapshot 1`] = `
<pm-about
  db={[Function MockDb]}
  test$={[Function ScalarObservable]}
>
  <h1>
    About
  </h1>
  <p>
    An Angular starter with all the tools necessary to create server rendered Angular applications.
  </p>
  
  <div>
    "test"
  </div>
</pm-about>
`;
